app/src/main/java/io/github/djunicode/canteenapp/BaseActivity.java

```diff
@@ -1,21 +1,72 @@
 package io.github.djunicode.canteenapp;
 
+import android.content.Intent;
 import android.content.SharedPreferences;
 import android.os.Bundle;
 import android.os.PersistableBundle;
 import android.support.annotation.Nullable;
 import android.support.v7.app.AppCompatActivity;
+import android.widget.Toast;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+import io.github.djunicode.canteenapp.ResponseObjects.SignInResponse;
+import okhttp3.Interceptor;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
 
 import static io.github.djunicode.canteenapp.LoginActivity.TOKEN_SHARED_PREFS_KEY;
 import static io.github.djunicode.canteenapp.LoginActivity.TOKEN_STRING;
 
 public abstract class BaseActivity extends AppCompatActivity {
 
     String tokenReturn;
+    protected Retrofit customRetrofit;
+    protected Boolean addToken = true;
+    private String toast;
 
     @Override
     protected void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        toast = this.getIntent().getStringExtra("toastMessage");
+        if (toast != null) {
+            Toast.makeText(getApplicationContext(), toast, Toast.LENGTH_SHORT).show();
+        }
+
+        OkHttpClient.Builder httpClient = new OkHttpClient.Builder();
```

Comment:
Why exactly do we need OkHttpClient? Is it something that cannot be done using Retrofit?

Reply:
I added that so i can increase the timeout, incase of slow connections the request will not get completed

Reply:
Okay :+1: